### ISSUE TITLE
Issue with template and .NET

### ISSUE THREAD

**Issue by ChristineLamontagne** [wet-boew/wet-boew#7401]
Hi!

In .NET, I can't use - in control's ID.  For exemple, if I try to add <asp:Button runat="server" ID="wb-srch-sub" CssClass="btn btn-default" Text="search" /> to do the search button at the top of the page, I get an error.  Unfortunatly, a lot of the CSS are based on the button name so if I change the name, the css no longer work properly.  Classes are fine but ID's must be changed so that it doesn't have such characters.

Thanks

**Comment by sgdowney** [comment]
You can use ClientIDMode="Static" to prevent .Net from changing the control id.

**Comment by ChristineLamontagne** [comment]
Hi!

The problem is not that .NET was changing the name, it’s that .NET is not allowing a name with a – character in it.  It refuses to compile.

Christine Lamontagne

**Comment by RobJohnston** [comment]
Agreed, the [Control.ID property] (https://msdn.microsoft.com/en-us/library/system.web.ui.control.id(v=vs.110).aspx) only allows "combinations of alphanumeric characters and the underscore character ( _ )".

You may be trying to overly dot-netify things. Can I point you to the [wet-boew-dotnet] (https://github.com/wet-boew/wet-boew-dotnet) variant to get you started?

**Comment by hsrudnicki** [comment]
Does this stay as an issue or can be this closed? Please update the status of this again, thank you
